lib/ModuleManager.cpp

```diff
@@ -836,6 +848,21 @@ void QoreModuleManager::loadModuleIntern(ExceptionSink& xsink, const char* name,
             }
         }
 
+        // check whether it is a module folder
+        QoreString modulePath(*w);
+        modulePath += QORE_DIR_SEP_STR;
+        modulePath += name;
+
+        if (QoreDir::folder_exists(modulePath, xsink)) {
+            mi = loadSeparatedModule(xsink, modulePath, name, pgm, reexport, pholder.release(), load_opt & QMLO_REINJECT ? mpgm : 0, load_opt);
```

Comment:
there is quite a lot pholder releases, is it really working?

Reply:
yes, it is, but when I was trying to simplify it a little, I got some error when trying to include module recursively, or in another complicated way, so I just copy paste it ...